WebClient/Scripts/temperatureControlView.js

```diff
@@ -0,0 +1,503 @@
+﻿
+L.Control.Temp = L.Control.extend({
+    options: {
+        collapsed: false,
+        position: 'topleft',
+        autoZIndex: true,
+        height: 150,
+        width: 150,
+    },
+
+    /// leaflet constructor
+    initialize: function (options) {
+        L.setOptions(this, options);
+        this.parentContainer = options.element;
+    },
+
+    onAdd: function (map) {
+        this.active = true;
+        this._initLayout();
+        this._map = map;
+        return this._container;
+    },
+
+    onRemove: function () {
+        this.active = false;
+    },
+
+    _initLayout: function () {
```

Comment:
This function is called from the **onAdd** and inside here a new div is created and added to the parent. The **onRemove** method doesn't remove the added div from the parent. This may cause problems when adding and removinfg the View for several times.